currency-rates/internal/handlers/subscribe.go

```diff
@@ -56,5 +59,32 @@ func (handler SubscribeHandler) CreateSubscription(w http.ResponseWriter, r *htt
 
 func isValidEmail(email string) bool {
 	return emailRegex.MatchString(email)
-  }
-  
\ No newline at end of file
+}
+  
+func (handler SubscribeHandler) DeactivateSubscription(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, "Form parse error", http.StatusBadRequest) 
+		return
+	}
+
+	formData := r.Form
+
+	email := formData.Get("email")
+	
+	err = handler.repo.DeactivateEmail(r.Context(), email)
+	responseStatus := http.StatusOK
+	
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			responseStatus = http.StatusNotFound
+		} else if errors.Is(err, errors.New("database unavailable")) {
```

Comment:
Seems very fragile. 
The logic should only depend on domain errors. If there some unhandeled erros like DB unvailable, you only return 500. And also good point to log unexpected erros, so you will know what's go wrong

Reply:
Done